value.go

```diff
@@ -194,9 +199,21 @@ func (v *MrbValue) Hash() *Hash {
 
 // String returns the "to_s" result of this value.
 func (v *MrbValue) String() string {
-	value := C.mrb_obj_as_string(v.state, v.value)
-	result := C.GoString(C.mrb_string_value_ptr(v.state, value))
-	return result
+	if v.IsDead() {
+		return ""
```

Comment:
Hm, I'd be open to that but I think its important we remain `Stringer` compliant for debugging reasons.

I think we may need a separate method for `(string, error)` and then `String` behaves as above. i.e. `String() string` may return something like "<error: %s>" but if you want to be able to distinguish errors from that literal string then you should call the other func that returns an error.

Hm... looking at the other methods like `Fixnum` their defined behavior is to panic in the case of errors. I suppose this makes sense because they must. Maybe we make a set of `TryX` methods like `TryString() (string, error)` ? Then in the future we might have `TryArray` (calls to_a if it exists), etc.?